spec/03-plugins/13-cors/01-access_spec.lua

```diff
@@ -204,20 +428,73 @@ for _, strategy in helpers.each_strategy() do
         },
       }
 
+      for i, testcase in ipairs(regex_testcases) do
+        local route = bp.routes:insert({
+          hosts = { "cors-regex-" .. i .. ".test" },
+        })
+
+        bp.plugins:insert {
+          name = "cors",
+          route = { id = route.id },
+          config = {
+            origins = testcase.origins,
+          }
+        }
+      end
+
       assert(helpers.start_kong({
         database   = strategy,
         nginx_conf = "spec/fixtures/custom_nginx.template",
       }))
 
-      proxy_client = helpers.proxy_client()
     end)
 
     lazy_teardown(function()
+      helpers.stop_kong(nil, true, true)
+    end)
+
+    before_each(function()
+      proxy_client = helpers.proxy_client()
+    end)
+
+    after_each(function()
       if proxy_client then proxy_client:close() end
-      helpers.stop_kong()
     end)
 
     describe("HTTP method: OPTIONS", function()
+
+      for i, testcase in ipairs(regex_testcases) do
+        local host = "cors-regex-" .. i .. ".test"
+        for origin, accept in sortedpairs(testcase.tests) do
+          it("given " .. origin .. ", " ..
+             inspect(testcase.origins) .. " will " ..
+             (accept and "accept" or "reject"), function()
+
+            local res = assert(proxy_client:send {
+              method  = "OPTIONS",
+              headers = {
+                ["Host"] = host,
+                ["Origin"] = origin,
+              }
+            })
+
+            assert.res_status(200, res)
+
+            if accept then
+              assert.equal("GET,HEAD,PUT,PATCH,POST,DELETE", res.headers["Access-Control-Allow-Methods"])
+              assert.equal(accept == true and origin or accept, res.headers["Access-Control-Allow-Origin"])
```

Comment:
`accept` can be `true`, `false` or a domain. That was ... surprising. I initially thought this tertiary operator was unnecessary (given the `if accept` clause two lines above).